### PR TITLE
teach copy-workflow to bump go.mod's version

### PR DESCRIPTION
**Pull request by mvdan**
(see commit message)

### CODE REVIEW

**Comment by mvdan** [comment]
testing job running here: https://github.com/protocol/.github/actions/runs/1071114583

**Comment by mvdan** [comment]
Oops, I'd left a debug print. Removed.

**Comment by mvdan** [comment]
I've also now added the docs on how the matrices work, as I'd mentioned earlier.

**Comment by mvdan** [comment]
Pinky promise I'm done force-pushing for now. I've pushed to testing again, to double-check, and it worked fine: https://github.com/protocol/.github/actions/runs/1071142101

**Review by marten-seemann — COMMENTED** [review]
This looks good! Lots of improvements to the original workflow!

File: .github/workflows/copy-workflow.yml [hunk 1]
```diff
@@ -38,10 +48,11 @@ jobs:
       with:
         # This should be the same Go version we use in the go-check workflow.
         # go mod tidy, go vet, staticcheck and gofmt might behave differently depending on the version.
-        go-version: "1.16.x"
+        stable: 'false'
+        go-version: "1.17.0-rc1"
```
[the remark this answers is not in the document]
Comment (mvdan) [reply .github/workflows/copy-workflow.yml:52]: Right, we have to change this when 1.17 comes out, even if just to remove `stable: false`. I don't think `1.17.x` works for pre-releases.

File: .github/workflows/dispatch.yml [hunk 2]
```diff
@@ -33,11 +33,19 @@ jobs:
...
+        #
+        #   [{"target": "repo1", "target": "repo2"}]
+        #
+        # The triggered copy-workflow jobs use that final array as their matrix.
+        # As such, we'll end up with one copy-workflow parallel job per target.
         cfg: ${{ fromJson(needs.matrix.outputs.targets) }}
     name: Start copy workflow (batch ${{ matrix.cfg.key }})
     steps:
-      - uses: peter-evans/repository-dispatch@ce5485de42c9b2622d2ed064be479e8ed65e76f4 # v1.1.3
+      - uses: benc-uk/workflow-dispatch@4c044c1613fabbe5250deadc65452d54c4ad4fc7 # v1
```
[the remark this answers is not in the document]
Comment (mvdan) [reply .github/workflows/dispatch.yml:47]: I'd missed that! done.

**Comment by mvdan** [comment]
all fixed up, I think

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**